File: connectors-native-sdk-java/src/main/java/com/snowflake/connectors/application/status/ConnectorStatus.java
/** Copyright (c) 2024 Snowflake Inc. */
package com.snowflake.connectors.application.status;

import java.util.Arrays;
import java.util.stream.Stream;

/** Connector status. */
public enum ConnectorStatus {

  /** Connector is being configured (during the wizard phase). */
  CONFIGURING,

  /** Connector is being resumed. */
  STARTING,

  /** Connector is configured and running. */
  STARTED,

  /** Connector is being paused. */
  PAUSING,

  /** Connector is paused. */
  PAUSED,

  /**
   * Connector is in an unspecified stated after encountering an irreversible error, it cannot be
   * actively used.
   */
  ERROR;

  /** Connector configuration status. */
  public enum ConnectorConfigurationStatus {

    /** Connector has just been installed. */
    INSTALLED,

    /** Prerequisites wizard phase has been completed. */
    PREREQUISITES_DONE,

    /** Connector configuration wizard phase has been completed. */
    CONFIGURED,

    /** Connection configuration wizard phase has been completed. */
    CONNECTED,

    /** Configuration finalization wizard phase has been completed. */
    FINALIZED;

    /**
     * Returns all possible configuration statuses excluding the ones provided.
     *
     * @param excludedStatuses configuration statuses to exclude
     * @return all possible configuration statuses excluding the ones provided
     */
    public static ConnectorConfigurationStatus[] getStatusesExcluding(
        ConnectorConfigurationStatus... excludedStatuses) {
      return Stream.of(ConnectorConfigurationStatus.values())
          .filter(it -> !Arrays.asList(excludedStatuses).contains(it))
          .toArray(ConnectorConfigurationStatus[]::new);
    }

    /**
     * Returns whether this status is before the provided configuration status during the wizard
     * phase.
     *
     * @param status configuration status to check
     * @return whether this status is before the provided configuration status during the wizard
     *     phase
     */
    public boolean isBefore(ConnectorConfigurationStatus status) {
      return this.ordinal() < status.ordinal();
    }

    /**
     * Returns whether this status is after the provided configuration status during the wizard
     * phase.
     *
     * @param status configuration status to check
     * @return whether this status is after the provided configuration status during the wizard
     *     phase
     */
    public boolean isAfter(ConnectorConfigurationStatus status) {
      return this.ordinal() > status.ordinal();
    }
  }
}

File: examples/connectors-native-sdk-example-java-github-connector/app/streamlit/native_sdk_api/connector_status.py
# Copyright (c) 2024 Snowflake Inc.

import json
import streamlit as st
from enum import Enum
from snowflake.snowpark.context import get_active_session


def load_connector_statuses():
    statuses = json.loads(
        get_active_session()
        .sql("CALL PUBLIC.GET_CONNECTOR_STATUS()")
        .collect()[0][0]
    )

    st.session_state["connector_status"] = ConnectorStatus[statuses.get("status")]
    st.session_state["configuration_status"] = ConfigurationStatus[
        statuses.get("configurationStatus")
    ]


def get_connector_status():
    return st.session_state["connector_status"]


def get_configuration_status():
    return st.session_state["configuration_status"]


def is_connector_in_error():
    return get_connector_status() == ConnectorStatus.ERROR


def is_connector_configuring():
    return get_connector_status() == ConnectorStatus.CONFIGURING


def is_connector_configured():
    return get_configuration_status() in [
        ConfigurationStatus.CONFIGURED,
        ConfigurationStatus.CONNECTED
    ]


def is_connection_configured():
    return get_configuration_status() in [
        ConfigurationStatus.CONNECTED
    ]


class ConnectorStatus(Enum):
    CONFIGURING = "CONFIGURING"
    STARTING = "STARTING"
    STARTED = "STARTED"
    PAUSING = "PAUSING"
    PAUSED = "PAUSED"
    ERROR = "ERROR"


class ConfigurationStatus(Enum):
    INSTALLED = "INSTALLED"
    PREREQUISITES_DONE = "PREREQUISITES_DONE"
    CONFIGURED = "CONFIGURED"
    CONNECTED = "CONNECTED"
    FINALIZED = "FINALIZED"
